lib/puppet/provider/cs_location/crm.rb

```diff
@@ -17,8 +17,29 @@
   # Decided to just go with relative.
   commands crm: 'crm'
 
+  # Path to the pacemakerd binary to check version for resource_discovery feature
+  # Decided to just go with relative.
+  commands pacemakerd: 'pacemakerd'
+
   mk_resource_methods
 
+  # we need to check if we run at least pacemakerd version 1.1.13 before enabling feature discovery
+  # see http://blog.clusterlabs.org/blog/2014/feature-spotlight-controllable-resource-discovery
+  begin
+    pacemakerd_version_string = pacemakerd('--version')
```

Comment:
Is this executed on the agent or during catalog compilation?

Reply:
It is executed on the agent.
(We are already using this code in our environment. If it would be executed during compilation it would be done on the puppetserver, where there is no pacemakerd installed and we then wouldn't have the discovery feature, which we have.)